docs/metrics-specification.md

```diff
@@ -6,31 +6,36 @@ This document outlines the specifications for a Prometheus exporter that collect
 ### Metrics Definition:
 
 _**PipelineRun Scheduling Duration:**_  
-The duration of time taken for a PipelineRun to be scheduled, calculated as the difference between the creation timestamp and the start time of the PipelineRun. 
+The duration of time in seconds taken for a PipelineRun to be scheduled, calculated as the difference between the creation timestamp and the start time of the PipelineRun. 
 
-_Metric Name:_ pipeline_run_scheduling_duration  
-_Labels:_ None  
-_Data Type:_ Gauge  
+_Metric Name:_ pipelinerun_scheduling_duration  
+_Labels:_ Minimally a `namespace` label.  If the `ENABLE_PIPELINERUN_SCHEDULED_DURATION_PIPELINENAME_LABEL` environment variable is set to `true` on the exporter deployment, the `pipelinename` label is set to the name of the Pipeline if its reference is set, otherwise the name of the PipelineRun.
+_Data Type:_ Histogram
 _Description:_ The time taken in seconds for a PipelineRun to be scheduled.
 
-_**PipelineRun Completion Duration:**_  
-The duration of time taken for a PipelineRun to complete, calculated as the difference between the start time and the completion time of the PipelineRun.  
-_Metric Name:_ pipeline_run_completion_duration  
-_Labels:_ None  
-_Data Type:_ Gauge  
-_Description:_ The time taken in seconds for a PipelineRun to complete.
+_**TaskRun Scheduling Duration:**_  
+The duration of time in seconds taken for a TaskRun to be scheduled, calculated as the difference between the creation timestamp and the start time of the TaskRun.
 
-### Data Collection:
-The data for these metrics will be collected from the status.startTime, status.completionTime, and metadata.creationTimestamp fields of PipelineRun objects in Pipeline Service.
+_Metric Name:_ taskrunrun_scheduling_duration  
+_Labels:_ Minimally a `namespace` label.  If the `ENABLE_TASKRUN_SCHEDULED_DURATION_TASKNAME_LABEL` environment variable is set to `true` on the exporter deployment, the `taskname` label is set to the name of the Task if its reference is set, otherwise the name of the TaskRun.
+_Data Type:_ Histogram
+_Description:_ The time taken in seconds for a TaskRun to be scheduled.
+
+
+_**Scheduling Duration of different TaskRuns with a PipelineRun:**_
+The time taken in milliseconds between the creation of the first TaskRun and the creation of its PipelineRun, followed by the duration in milliseconds between the completion of a preceding TaskRun and the creation of the following TaskRun.  This metrics current assumes sequential TaskRuns, and the parallel TaskRuns within a PipelineRun are not employed.  At this time, this metric is disabled by default, but can be enabled by setting the `ENABLE_GAP_METRIC` to `true`.
```

Comment:
Nitpick:
```suggestion
The time taken in milliseconds between the creation of the first TaskRun and the creation of its PipelineRun, followed by the duration in milliseconds between the completion of a preceding TaskRun and the creation of the following TaskRun.  This metric current assumes sequential TaskRuns, and the parallel TaskRuns within a PipelineRun are not employed.  At this time, this metric is disabled by default, but can be enabled by setting the environment variable `ENABLE_GAP_METRIC` to `true`.
```

Reply:
Maybe also `This metric current assumes` -> `This metric currently assumes`.